authcrypt/main.cpp

```diff
@@ -53,132 +39,177 @@ static const unsigned char secret_key[16] = {
     0xb9, 0xa0, 0xb8, 0xe9, 0x87, 0xb8, 0xc1, 0x72,
 };
 
-static int example(void)
-{
-    /* message that should be protected */
-    const char message[] = "Some things are better left unread";
-    /* metadata transmitted in the clear but authenticated */
-    const char metadata[] = "eg sequence number, routing info";
-    /* ciphertext buffer large enough to hold message + nonce + tag */
-    unsigned char ciphertext[128] = { 0 };
-    int ret;
-
-    mbedtls_printf("\r\n\r\n");
-    print_hex("plaintext message", (unsigned char *) message, sizeof message);
-
-    /*
-     * Setup random number generator
-     * (Note: later this might be done automatically.)
-     */
-    mbedtls_entropy_context entropy;    /* entropy pool for seeding PRNG */
-    mbedtls_ctr_drbg_context drbg;      /* pseudo-random generator */
-
-    mbedtls_entropy_init(&entropy);
-    mbedtls_ctr_drbg_init(&drbg);
-
-    /* Seed the PRNG using the entropy pool, and throw in our secret key as an
-     * additional source of randomness. */
-    ret = mbedtls_ctr_drbg_seed(&drbg, mbedtls_entropy_func, &entropy,
-                                       secret_key, sizeof secret_key);
-    if (ret != 0) {
-        return 1;
+/* Message that should be protected */
+static const char message[] = "Some things are better left unread";
+/* Metadata transmitted in the clear but authenticated */
+static const char metadata[] = "eg sequence number, routing info";
```

Comment:
Any particular reason for leaving ```message```, ```metadata``` and ```secret_key``` outside the class. If this data belongs to this class then it should be encapsulated inside this class.